src/PactNet.Native/PactVerifier.cs

```diff
@@ -64,6 +68,36 @@ public IPactVerifier ServiceProvider(string providerName, Uri pactUri)
             return this;
         }
 
+        /// <summary>
+        /// Set the provider details
+        /// </summary>
+        /// <param name="providerName">Name of the provider</param>
+        /// <param name="pactUri">URI of the running service</param>
+        /// <param name="messagingRoute">Route where the messaging interaction is going to be sent</param>
+        /// <returns>Fluent builder</returns>
+        public IPactVerifier MessagingServiceProvider(string providerName, Uri pactUri, string messagingRoute)
```

Comment:
This makes the API kind of weird because you could technically call both kids of service provider method, and then I'm not sure what would happen.

Reply:
I'm confused with your comment. At first, I had a MessagingPactVerifier which granted was customizing the implementation and was not really solid friendly. So I added a new method to the general verifier interface.
Are you referring to putting some kind of guard once you call one of the service provider methods?

Reply:
I think we should take care of it it in your oncoming PR